mlflow_client/core.py

```diff
@@ -0,0 +1,235 @@
+import uuid
+import os
+import sys
+
+from abc import ABC, abstractmethod
+from enum import Enum
+from contextlib import contextmanager
+
+
+class MLFrameworks(Enum):
+    """
+    Enum to choose between ml frameworks
+    """
+    PYTHON = 0
+    SCIKIT_LEARN = 1
+    KERAS = 2
+    PYTORCH = 3
+
+
+def if_run_active(method):
+    """
+    Prevent logging if no active run is found in the current backend
+    """
+    def wrapper(self, *args):
+        if self._run_started:
+            return method(self, *args)
+        else:
+            raise ValueError("Logging method called without any active run")
+    return wrapper
+
+
+def get_auto_backend():
+    """
+    Returns a backend automatically
+    LocalBackend if mlflow is not found
+    MLFLow Local if mlflow is found
+    MLFlow Server if mlflow is found and MLFLOW_TRACKING_URI defined in env variables
+    """
+    try:
```

Comment:
add a logger message to display which environment is chosen

Reply:
i put a logging warning so that every logger with default config could log this message, should I put it to info ?